lib/client/redirect.go

```diff
@@ -202,6 +231,11 @@ func (rd *Redirector) callback(w http.ResponseWriter, r *http.Request) (*auth.SS
 		return nil, trace.NotFound("path not found")
 	}
 
+	if r.URL.Query().Has("err") {
+		err := r.URL.Query().Get("err")
```

Comment:
Should this check that `err` isn't empty also?

Reply:
If the `err` is present but empty there isn't much we can do differently here. No way to fail even harder...